todoman/interactive.py

```diff
@@ -0,0 +1,583 @@
+import urwid
+from .ui import TodoEditor, EditState
+from .model import Todo
+
+import logging
+logging.basicConfig()
+
+logger = logging.getLogger()
+
+class TodomanItem(urwid.CheckBox):
+    '''
+    Class to contain a single todo item in a ListBox.
+    '''
+
+    def __init__(self, filename, database, labelmaker):
+        '''
+        Create a TodomanItem instance based on the filename and the associated
+        Database that was provided. By providing the filename instead of the
+        Todo itself, we do not need to check if the provided todo is indeed in
+        in the given Database. The labelmaker is a function that turns a given
+        TodomanItem (i.e. self) into a string representation that is suitable
+        for its context.
+
+        (TodomanItem, str, Database, function) -> None
+        '''
+        self.filename = filename
+        self.database = database
+        try:
```

Comment:
Might as well have the params `database, todo` instead of `database, filename` and avoid this logic.